lang/ghc/Portfile

```diff
@@ -109,47 +84,63 @@ compiler.cpath      /usr/bin/gcc
 compiler.library_path \
                     /usr/lib:${prefix}/lib
 
+# use these to specify python versions, python3 required
+set python3_version 3.7
+set python3_version_nickname \
+                    [join [lrange [split ${python3_version} .] 0 1] {}]
+
+set srcpath ${workpath}/src
+
 # Note: ghc has DESTDIR; however, specifying --prefix=${prefix} installs
 # binaries into ${prefix}, not DESTDIR.  Work around this by setting
 # --prefix=${destroot}${prefix} and not setting DESTDIR
 
-if {![variant_isset "bootstrap"]} {
+if { [variant_isset "prebuilt"] } {
     set bootstrap_dir ${destroot}${prefix}
 } else {
     set bootstrap_dir ${workpath}/bootstrap
 
-    pre-configure {
-        xinstall -m 0755 -d ${bootstrap_dir}
-    }
-}
+    distfiles-append \
+                    ${distname}-src${extract.suffix}
+    checksums-append \
+                    ${distname}-src${extract.suffix} \
+                    rmd160  2640736ab93348a6f9550d914d811c0e62b77e50 \
+                    sha256  4d4aa1e96f4001b934ac6193ab09af5d6172f41f5a5d39d8e43393b9aafee361 \
+                    size    19092168
 
-# install the initial bootstrap compiler to destroot
-configure.pre_args  --prefix=${bootstrap_dir}
+    depends_build-append \
+                    port:autoconf \
+                    port:automake \
+                    port:python${python3_version_nickname} \
+                    port:py${python3_version_nickname}-sphinx \
+                    port:texlive \
+                    port:texlive-fonts-extra \
+                    port:texlive-fonts-recommended
 
-build.target        install
+    depends_lib-append \
```

Comment:
Are those really required at runtime, or only for the build?

Reply:
There are references all throughout the library files (see `find` comment above), so my best guess is that this really is a runtime dependency.

Reply:
I did some checking. I believe the library symbol references are for downloaded/compiled code done by `ghc-cabal`, and these are `build` dependencies, not `library`. I've changed this.